operations/power_management/Prepare_the_System_for_Power_Off.md

```diff
@@ -25,62 +25,112 @@ HPE Cray EX System Admin Toolkit (SAT) product stream documentation (`S-8031`) f
 
    See the "SAT Authentication" section in the HPE Cray EX System Admin Toolkit (SAT) product stream documentation (`S-8031`) for instructions on how to acquire a SAT authentication token.
 
-1. (`ncn-mw#`) Determine which Boot Orchestration Service \(BOS\) templates to use to shut down compute nodes and UANs.
+1. Ensure `/root/.bashrc` has proper handling of `kubectl` commands on all master and worker nodes.
 
-   There will be separate session templates for UANs and computes nodes.
+   **Important:** During the process of shutting down the system, there will be a point when `kubelet` will be stopped on all the master and worker
+   nodes. Once `kubelet` has been stopped, any `kubectl` command on any master or worker node may not work as expected and may have a long timeout before
+   failing.
 
-    1. List all the session templates.
+   This issue can cause a slowdown for these `sat` commands which `ssh` from the `sat` pod to `ncn-m001` and the
+   other nodes because the `ssh` will execute commands from `/root/.bashrc`.
 
-       If it is unclear what session template is in use, proceed to the next substep.
+      * Commands affected during the power down
+         * `sat bootsys shutdown --stage platform-services`
+         * `sat bootsys shutdown --stage ncn-power`
 
-       ```bash
-       cray bos v1 sessiontemplate list
-       ```
+      * Commands affected during the power up
+         * `sat bootsys boot --stage ncn-power`
+         * `sat bootsys boot --stage platform-services`
 
-    1. Find the xname with `sat status`.
+      1. Here is a sample command in `/root/.bashrc` which sets an environment variable using the output from `kubectl` which has the problem.
 
-       ```bash
-       sat status | grep "Compute\|Application"
-       ```
+         ```bash
+         export DOMAIN=$(kubectl get secret site-init -n loftsman -o jsonpath='{.data.customizations\.yaml}'|base64 -d | grep "external:")
+         ```
 
-       Example output:
+      1. This shows one way to correct that sample command so the environment variable will be set when `kubelet` is available and will skip setting the variable when `kubelet` is not available.
 
-       ```text
-       | x3000c0s19b1n0 | Node | 1        | On    | OK   | True    | X86  | River | Compute     | Sling    |
-       | x3000c0s19b2n0 | Node | 2        | On    | OK   | True    | X86  | River | Compute     | Sling    |
-       | x3000c0s19b3n0 | Node | 3        | On    | OK   | True    | X86  | River | Compute     | Sling    |
-       | x3000c0s19b4n0 | Node | 4        | On    | OK   | True    | X86  | River | Compute     | Sling    |
-       | x3000c0s27b0n0 | Node | 49169248 | On    | OK   | True    | X86  | River | Application | Sling    |
-       ```
+         ```bash
+         if [ ! `systemctl is-active kubelet | grep inactive` ] ; then
```

Comment:
The sample is functional as it is, with "if not inactive; then set variable".  This is shorter than "if active; then echo active ; else set variable".

Reply:
Are you sure that your test is actually valid? There are backticks around `systemctl is-active kubelet | grep inactive`, which just means bash will expand that to the empty string when kubelet is active or the string "inactive" when it isn't. Then your test becomes `[ ! inactive ]`, and I don't think that's what you intended.

I also think my suggestion is shorter. What am I missing?

Reply:
Pulled both sample chunks of code into shell script (instead of doing interactively) and tested again.

Switching to Ryan's method.

```bash
-         if [ ! `systemctl is-active kubelet | grep inactive` ] ; then
+         if [ systemctl is-active -q kubelet ; then
                  export DOMAIN=$(kubectl get secret site-init -n loftsman -o jsonpath='{.data.customizations\.yaml}'|base64 -d | grep "external:")
          fi
```